### PR TITLE
Docs/improvements

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- ~~Added/Updated related tests.~~

## Short description of the change(s)

This fixes a bunch of issues related to documentation.

### CODE REVIEW

**Comment by molant** [comment]
I'm renaming a section in the docs entirely that all the packages point to (the .hintrc one). We also have a problem with the documentation in the site not showing the main page for subsections (and that page is now a main one). I'm going to work on that thing next but still will need to update the links from all the packages. I still haven't decide how to solve that because we need to make sure everything rests accessible. I'll continue the discussion over there so maybe review and wait until that one is merged?

**Comment by molant** [comment]
Once webhintio/webhint.io#549 is merged, the content will look like this:

image

File: packages/hint/docs/contributor-guide/getting-started/events.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 Information is shared internally via `event`s. `connector`s and `parser`s can
 create them, while `parser`s and `hint`s consume them.
 The following is a list of all the events common to all `connector`s, with
-their signature, and the `interface` they implement.
+their signature, and the `interface` they implement. The exception is the
+`local connector` that needs the `html parser` to emit these events:
```
Comment (alrra) [inline packages/hint/docs/contributor-guide/getting-started/events.md]: HTML parser

File: packages/hint/docs/contributor-guide/guides/create-custom-hint.md [hunk 2]
```diff
@@ -65,13 +64,17 @@ The next main thing to do is to populate the `validate-footer.ts` and `test.ts`
...
+1. Install all the dependencies.
+2. Build the project, which compiles the TypeScript to JavaScript in the `dist`
    folder (support for creating JavaScript hints via webhint's CLI will be
    added in a future release, but you can manually do it now).
 
-Do remember that since a compilation is needed, you have to run build every time
-after you make changes to the .ts files. Or alternatively, you run `npm run
+**Note for core hints**: Instead of running `npm run init` run `yarn` and
+`yarn build`. `webhint` is a monorepo built on top of `yarn` workspaces and this
+is the proper way to link the the modules.
```
Comment (alrra) [inline packages/hint/docs/contributor-guide/guides/create-custom-hint.md]: Remove extra `the`.

File: packages/hint/docs/user-guide/concepts/parsers.md [hunk 3]
```diff
@@ -16,40 +16,35 @@ configuration file:
...
   parser built on top of `ESLint` so hints for analyzing `JavaScript`
   files can be built.
 
+* [`html`][@hint/parser-html]: An `HTML` parser built on top of `jsdom`.
+  This parser is only necessary if you are using the `local connector`
+  and analyzing local `HTML` files. Otherwise the related `HTML` events
+  are taken care directly by the other `connector`s.
+
+* [`manifest`][@hint/parser-manifest]: A parser that validates if a
+  `webmanifest` is valid and emit information related to it.
```
Comment (alrra) [inline packages/hint/docs/user-guide/concepts/parsers.md]: > web app manifest file is valid...

File: packages/hint/docs/user-guide/configuring-webhint/summary.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# Configuring webhint
+
+There are 2 ways in which you can configure `webhint`:
+
+1. Via a `.hintrc` file.
+1. Adding a `webhintConfig` property in your `package.json`.
```
Comment (alrra) [inline packages/hint/docs/user-guide/configuring-webhint/summary.md]: It's `hintConfig`, not `webhintConfig`.

File: packages/hint/docs/user-guide/configuring-webhint/summary.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+{
+    "name": "project name",
+    "dependencies": {
+        ...
+    },
+    "scripts": {
+        ...
+    },
+    ...
+    "webhintConfig": {
```
Comment (alrra) [inline packages/hint/docs/user-guide/configuring-webhint/summary.md]: `hintConfig`

File: packages/hint/docs/user-guide/configuring-webhint/summary.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+The `severity` of a `hint` can be:
+
+* `off`: The `hint` will not be run. This is the same as deleting it
+  from the `.hintrc`.
+* `warning`: If the `hint` finds any issue it will be reported but the
+  exit code will be `0`.
+* `error`: If the `hint` finds any issue it will be reported and the
+  exit code will be `1`.
+
```
Comment (alrra) [inline packages/hint/docs/user-guide/configuring-webhint/summary.md]: Remove extra empty line.

File: packages/hint/docs/contributor-guide/getting-started/events.md (hunk 1 above)
Comment (molant) [reply packages/hint/docs/contributor-guide/getting-started/events.md]: You can use the newest GitHub feature for this changes 
image 😋

Comment (alrra) [reply packages/hint/docs/contributor-guide/getting-started/events.md]: Not on mobile.

**Comment by molant** [comment]
@alrra changed done and force pushed so it's in the right commits.

File: packages/hint/docs/contributor-guide/getting-started/events.md [hunk 7]
```diff
@@ -3,7 +3,8 @@
 Information is shared internally via `event`s. `connector`s and `parser`s can
 create them, while `parser`s and `hint`s consume them.
 The following is a list of all the events common to all `connector`s, with
-their signature, and the `interface` they implement.
+their signature, and the `interface` they implement. The exception is the
+`local connector` that needs the `HTML parser` to emit these events:
```
Comment (antross) [inline packages/hint/docs/contributor-guide/getting-started/events.md]: FYI we have some adjacent edits in my typed events PR:
https://github.com/webhintio/hint/pull/1397/files#diff-a9286171f2cf5bcdf385128b16fa25ad

File: packages/hint/docs/user-guide/concepts/parsers.md [hunk 8]
```diff
@@ -16,40 +16,35 @@ configuration file:
 
 ## List of official `parser`s
 
+* [`css`][@hint/parser-css]: A `CSS` parser built on top of
```
Comment (antross) [inline packages/hint/docs/user-guide/concepts/parsers.md]: FYI, I also updated this list in my typed events PR:
https://github.com/webhintio/hint/pull/1397/files#diff-ac4792357afd3a5de0c4ab23c27aee72

File: packages/hint/docs/user-guide/concepts/parsers.md [hunk 9]
```diff
@@ -16,40 +16,35 @@ configuration file:
...
+  [PostCSS][postcss] so hints can analyze `CSS` files.
+
 * [`javascript`][@hint/parser-javascript]: A `JavaScript`
   parser built on top of `ESLint` so hints for analyzing `JavaScript`
   files can be built.
 
+* [`html`][@hint/parser-html]: An `HTML` parser built on top of `jsdom`.
```
Comment (antross) [inline packages/hint/docs/user-guide/concepts/parsers.md]: Did you intend to list the parsers alphabetically? If so, this should be before `javascript`.

File: packages/hint/docs/user-guide/concepts/parsers.md [hunk 10]
```diff
@@ -16,40 +16,35 @@ configuration file:
 
 ## List of official `parser`s
```
Comment (antross) [inline packages/hint/docs/user-guide/concepts/parsers.md]: Did you want to include `babel-config` here? (I did in my PR):
```suggestion

* [`babel-config`][@hint/parser-babel-config] A `Babel configuration` parser
  which validates the provided `json` so hints analyzing `.babelrc` files can
  be built.

```

File: packages/hint/docs/user-guide/concepts/parsers.md [hunk 11]
```diff
@@ -16,40 +16,35 @@ configuration file:
...
-To create a hint that understands a TypeScript configuration you will need
-to use the event `parse::typescript-config` emiited by the
-[`typescript-config`][@hint/parser-typescript-config].
-This event is of type `TypeScriptConfigParse` wich has the following
-information:
-
-* `resource`: the parsed resource.
-* `config`: an object with a valid the configuration.
 
 <!-- Link labels: -->
```
Comment (antross) [inline packages/hint/docs/user-guide/concepts/parsers.md]: If you decide to include `babel-config` we'll need the link too:
```suggestion

[@hint/parser-babel-config]: https://npmjs.com/package/@hint/parser-babel-config/
```

File: packages/hint/docs/user-guide/configuring-webhint/summary.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+In both cases, the format used is the same. The following is an example
+for a `.hintrc`:
+
+```json
+{
+    "connector": {
+        "name": "connectorName"
+    },
+    "formatters": ["formatterName"],
+    "pasers": ["parserName"],
```
Comment (antross) [inline packages/hint/docs/user-guide/configuring-webhint/summary.md]:
```suggestion
    "parsers": ["parserName"],
```

File: packages/hint/docs/user-guide/configuring-webhint/summary.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+    "scripts": {
+        ...
+    },
+    ...
+    "hintConfig": {
+        "connector": {
+            "name": "connectorName"
+        },
+        "formatters": ["formatterName"],
+        "pasers": ["parserName"],
```
Comment (antross) [inline packages/hint/docs/user-guide/configuring-webhint/summary.md]:
```suggestion
        "parsers": ["parserName"],
```

**Comment by molant** [comment]
> do we call out the ability to run against the filesystem somewhere

Will open an issue for that as I think it needs some thinking and probably more than just a few word (short video will be nice...).

File: packages/hint/docs/user-guide/concepts/parsers.md (hunk 10 above)
Comment (molant) [reply packages/hint/docs/user-guide/concepts/parsers.md]: Done

**Comment by molant** [comment]
@antross changes done.

**Pull request closed without merging**